src/plays/hangman-game/components/Keyboard.tsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect } from 'react';
+import React from 'react';
+import styled from 'styled-components';
+
+const KEYS = [
+  'a',
+  'b',
+  'c',
+  'd',
+  'e',
+  'f',
+  'g',
+  'h',
+  'i',
+  'j',
+  'k',
+  'l',
+  'm',
+  'n',
+  'o',
+  'p',
+  'q',
+  'r',
+  's',
+  't',
+  'u',
+  'v',
+  'w',
+  'x',
+  'y',
+  'z'
+];
+
+interface KeyboardProps {
+  correctLetters: string[];
+  incorrectLetters: string[];
+  addGuessedLetter: (letter: string) => void;
+  disabled?: boolean;
+}
+
+export default function Keyboard({
+  correctLetters,
+  incorrectLetters,
+  addGuessedLetter,
+  disabled = false
+}: KeyboardProps) {
+  useEffect(() => {
+    const handleKeyPress = (event: KeyboardEvent) => {
+      const key = event.key.toLowerCase();
+      if (
+        !disabled &&
+        KEYS.includes(key) &&
+        !correctLetters.includes(key) &&
+        !incorrectLetters.includes(key)
+      ) {
+        addGuessedLetter(key);
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyPress);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyPress);
+    };
+  }, [correctLetters, incorrectLetters, addGuessedLetter, disabled]);
+
+  return (
+    <KeyContainer>
+      <KeyboardGrid>
+        {KEYS.map((key) => {
+          const active = correctLetters.includes(key);
+          const inActive = incorrectLetters.includes(key);
+
+          return (
+            <Key
+              active={active}
+              disabled={active || inActive || disabled}
+              inActive={inActive}
+              key={key}
+              onClick={() => addGuessedLetter(key)}
+            >
+              {key}
+            </Key>
+          );
+        })}
+      </KeyboardGrid>
+    </KeyContainer>
+  );
+}
+
+const KeyContainer = styled.div`
+  align-self: stretch;
+`;
+
+const KeyboardGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(50px, 1fr));
+  gap: 0.5rem;
+  width: 100%;
+`;
```

Comment:
Move all styled-components to `styles.css` and import & use them.

Benefits 
- the JS logic and styles will be separate. So, easier to debug specific problems.
- the respective lines on code on each files will be less

Reply:
@sudipkundu999  done!